/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:234: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler()
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:234: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler()
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:234: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler()
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:234: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler()
/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py:624: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
Simulator constructed!
Dataset created!
Simulator constructed!
Dataset created!
Simulator constructed!
Dataset created!
Simulator constructed!
Dataset created!
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/simulator.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  p = torch.tensor(params, dtype=torch.float32, device=self.device)
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:247: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast():
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:247: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast():
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:247: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast():
/pscratch/sd/k/katiekee/PDFParameterInference/cl.py:247: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast():
Epoch 1, Loss: 46.7860
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    prepare(preparation_data)
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    main_content = runpy.run_path(main_path,
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 122, in spawn_main
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 131, in _main
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    prepare(preparation_data)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 246, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    _fixup_main_from_path(data['init_main_from_path'])
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/multiprocessing/spawn.py", line 297, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
    main_content = runpy.run_path(main_path,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 13, in <module>
    from datasets import *
  File "/pscratch/sd/k/katiekee/PDFParameterInference/datasets.py", line 86, in <module>
    class DISDataset(IterableDataset):
                     ^^^^^^^^^^^^^^^
NameError: name 'IterableDataset' is not defined
[rank0]:[W622 15:25:14.428242643 ProcessGroupNCCL.cpp:1496] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
[rank2]:[W622 15:25:15.270122432 CudaIPCTypes.cpp:16] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]
[rank0]:[W622 15:25:15.271979771 CudaIPCTypes.cpp:16] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]
[rank1]:[W622 15:25:15.273316693 CudaIPCTypes.cpp:16] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]
[rank3]:[W622 15:25:15.275848905 CudaIPCTypes.cpp:16] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]
W0622 15:25:24.614000 2110346 site-packages/torch/multiprocessing/spawn.py:169] Terminating process 2110373 via signal SIGTERM
W0622 15:25:24.615000 2110346 site-packages/torch/multiprocessing/spawn.py:169] Terminating process 2110374 via signal SIGTERM
W0622 15:25:24.615000 2110346 site-packages/torch/multiprocessing/spawn.py:169] Terminating process 2110383 via signal SIGTERM
Traceback (most recent call last):
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 339, in <module>
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/multiprocessing/spawn.py", line 340, in spawn
    return start_processes(fn, args, nprocs, join, daemon, start_method="spawn")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/multiprocessing/spawn.py", line 296, in start_processes
    while not context.join():
              ^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/multiprocessing/spawn.py", line 215, in join
    raise ProcessRaisedException(msg, error_index, failed_process.pid)
torch.multiprocessing.spawn.ProcessRaisedException: 

-- Process 2 terminated with the following error:
Traceback (most recent call last):
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 1251, in _try_get_data
    data = self._data_queue.get(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/queue.py", line 180, in get
    self.not_empty.wait(remaining)
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/threading.py", line 359, in wait
    gotit = waiter.acquire(True, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/_utils/signal_handling.py", line 73, in handler
    _error_if_any_worker_fails()
RuntimeError: DataLoader worker (pid 2125013) exited unexpectedly with exit code 1. Details are lost due to multiprocessing. Rerunning with num_workers=0 may give better error trace.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/multiprocessing/spawn.py", line 90, in _wrap
    fn(i, *args)
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 321, in main_worker
  File "/pscratch/sd/k/katiekee/PDFParameterInference/cl.py", line 240, in train
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 708, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 1458, in _next_data
    idx, data = self._get_data()
                ^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 1410, in _get_data
    success, data = self._try_get_data()
                    ^^^^^^^^^^^^^^^^^^^^
  File "/global/common/software/nersc9/pytorch/2.6.0/lib/python3.12/site-packages/torch/utils/data/dataloader.py", line 1264, in _try_get_data
    raise RuntimeError(
RuntimeError: DataLoader worker (pid(s) 2125013) exited unexpectedly

